.tekton/tasks/aro-deploy-only-task.yaml

```diff
@@ -0,0 +1,270 @@
+      type: string
+      description: "ARO location"
+      default: "northeurope"
+    - name: aroClusterName
+      type: string
+      description: "ARO cluster name"
+    - name: aroVersion
+      type: string
+      description: "ARO version"
+      default: "4.15.35"
+    - name: azureSecretName
+      type: string
+      description: "Name of the Kubernetes Secret containing Azure service principal credentials, ARO resource group, and ARO domain"
+      default: "azure-sp-secret"
+    - name: pullSecretName
+      type: string
+      description: "Name of the Kubernetes Secret containing Red Hat pull secret"
+      default: "redhat-pull-secret"
+  results:
+    - name: aro-console-url
+      description: "ARO cluster console URL"
+    - name: aro-api-url
+      description: "ARO cluster API server URL"
+    - name: aro-username
+      description: "ARO cluster admin username"
+    - name: aro-password
+      description: "ARO cluster admin password"
+    - name: kubeconfig
+      description: "ARO cluster kubeconfig content"
+  workspaces:
+    - name: source
+  steps:
+    - name: deploy-aro-only
+      image: registry.access.redhat.com/ubi9/ubi
+      workingDir: $(workspaces.source.path)
+      env:
+        - name: CLIENT_ID
+          valueFrom:
+            secretKeyRef:
+              name: $(params.azureSecretName)
+              key: CLIENT_ID
+        - name: CLIENT_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: $(params.azureSecretName)
+              key: CLIENT_SECRET
+        - name: TENANT_ID
+          valueFrom:
+            secretKeyRef:
+              name: $(params.azureSecretName)
+              key: TENANT_ID
+        - name: ARO_RESOURCE_GROUP
+          valueFrom:
+            secretKeyRef:
+              name: $(params.azureSecretName)
+              key: ARO_RESOURCE_GROUP
+        - name: ARO_DOMAIN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.azureSecretName)
+              key: ARO_DOMAIN
+        - name: PULL_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pullSecretName)
+              key: PULL_SECRET
+        - name: ARO_LOCATION
+          value: $(params.aroLocation)
+        - name: ARO_CLUSTER_NAME
+          value: $(params.aroClusterName)
+        - name: ARO_VERSION
+          value: $(params.aroVersion)
+      script: |
+        #!/bin/bash
+        set -euo pipefail
+        
+        echo "🚀 Starting ARO cluster deployment (no PostgreSQL/Redis)..."
+        
+        echo "🔧 Using configuration:"
+        echo "  ARO_CLUSTER_NAME: ${ARO_CLUSTER_NAME}"
+        echo "  ARO_LOCATION: ${ARO_LOCATION}"
+        echo "  ARO_VERSION: ${ARO_VERSION}"
+        
+        # Install required tools
+        echo "📦 Installing required tools..."
+        # Install make, libicu, jq, and python3-pip first (needed for makefile operations and Ansible)
+        dnf install -y make libicu jq python3-pip
```

Comment:
The commands below are common in most of the files in the @tasks folder, these can be put into a shared initialization script to make it more abstract.

```
dnf install -y make libicu jq python3-pip
rpm --import https://packages.microsoft.com/keys/microsoft.asc
dnf install -y https://packages.microsoft.com/config/rhel/9.0/packages-microsoft-prod.rpm
dnf install -y azure-cli-2.59.0
curl -L https://mirror.openshift.com/pub/openshift-v4/clients/ocp/stable/openshift-client-linux.tar.gz | tar xz
pip3 install ansible kubernetes
ansible-galaxy collection install -r ansible/requirements.yml
```